evm-template/runtime/src/configs/xcm_config.rs

```diff
@@ -372,3 +390,221 @@ parameter_types! {
         ].into()
     };
 }
+
+parameter_type_with_key! {
+    pub ParachainMinFee: |location: Location| -> Option<u128> {
+        match (location.parents, location.first_interior()) {
+            // Polkadot AssetHub fee
+            (1, Some(Parachain(1000u32))) => Some(50_000_000u128),
+            _ => None,
+        }
+    };
+}
+
+// Our currencyId. We distinguish for now between SelfReserve, and Others, defined by their Id.
+#[derive(Clone, Eq, Debug, PartialEq, Ord, PartialOrd, Encode, Decode, TypeInfo)]
+pub enum CurrencyId {
+    // Our native token
+    SelfReserve,
+    // Assets representing other chains native tokens
+    ForeignAsset(AssetId),
+    // Erc20 token
+    Erc20 { contract_address: H160 },
+}
+
+impl AccountIdToCurrencyId<AccountId, CurrencyId> for Runtime {
+    fn account_to_currency_id(account: AccountId) -> Option<CurrencyId> {
+        Some(match account {
+            // pallet_balances only store information about the native token
+            // if the account is found in the balances pallet, it's our native token
+            a if pallet_balances::Account::<Runtime>::contains_key(a) => CurrencyId::SelfReserve,
```

Comment:
@alistair-singh 

This is the last change we needed to make imo.
When you have the chance, if you can review it once more, we can merge it 🚀

Reply:
I don't know about these lines, but it looks like it is converting a contract address to a currency. This doesn't have much to do with bridging so I cannot comment here.